scripts/firedrake-install

```diff
@@ -1599,7 +1617,10 @@ else:
         petsc_changed = git_update("petsc", deps["petsc"])
     else:
         petsc_changed = False
-    petsc4py_changed = git_update("petsc4py", deps["petsc4py"])
+    if use_petsc4py_in_petsc:
+        petsc4py_changed = petsc_changed
+    else:
+        petsc4py_changed = git_update("petsc4py", deps["petsc4py"])
```

Comment:
`petsc4py_changed = petsc_changed or git_update(...)` I think in case the petsc ABI changes.